sklearn/decomposition/pca.py

```diff
@@ -96,22 +102,20 @@ def _infer_dimension_(spectrum, n_samples, n_features):
     return ll.argmax()
 
 
-class PCA(BaseEstimator, TransformerMixin):
+class PCA(_BasePCA):
     """Principal component analysis (PCA)
 
     Linear dimensionality reduction using Singular Value Decomposition of the
-    data and keeping only the most significant singular vectors to project the
-    data to a lower dimensional space.
-
-    This implementation uses the scipy.linalg implementation of the singular
-    value decomposition. It only works for dense arrays and is not scalable to
-    large dimensional data.
+    data to project it to a lower dimensional space.
 
-    The time complexity of this implementation is ``O(n ** 3)`` assuming
-    n ~ n_samples ~ n_features.
+    It uses the scipy.linalg LAPACK implementation of the SVD or a randomized
+    SVD by the method of Halko et al. 2009, wdepending on which is the most
+    efficient.
 
     Read more in the :ref:`User Guide <PCA>`.
 
+    .. versionchanged:: 0.18
```

Comment:
I don't think we should use `versionchanged` in the docstring of the class itself, only in the docstring of the parameters that are impacted by the change. Otherwise it's redundant.

Note: `versionadded` is perfectly fine to me though.